docs/_docs/best-practices/compute/compute.md

```diff
@@ -88,26 +76,60 @@ resources
 
-This query analyzes Virtual Machine Scale Sets (VMSS) in your Azure environment based on their SKU, spot VM priority, and priority mix policy. It provides insights for cost optimization and resource management strategies.
+### Query: List Virtual Machines deallocated
+
+This Azure Resource Graph (ARG) query identifies Virtual Machines (VMs) in your Azure environment that are in the 'deallocated' state. It retrieves details about their power state, location, resource group, and subscription ID.
 
 <h4>Category</h4>
 
-Resource management
+Waste reduction
 
 <h4>Query</h4>
 
 ```kql
 resources
-| where type =~ 'microsoft.compute/virtualmachinescalesets'
-| extend SpotVMs = tostring(properties.virtualMachineProfile.priority)
-| extend SpotPriorityMix = tostring(properties.priorityMixPolicy)
-| extend SKU = tostring(sku.name)
-| extend resourceGroup = strcat('/subscriptions/', subscriptionId, '/resourceGroups/', resourceGroup)
-| project id, SKU, SpotVMs, SpotPriorityMix, subscriptionId, resourceGroup, location
+| where type =~ 'microsoft.compute/virtualmachines'
+    and tostring(properties.extended.instanceView.powerState.displayStatus) == 'VM deallocated'
+| where resourceGroup in ({ResourceGroup})
+| extend  PowerState=tostring(properties.extended.instanceView.powerState.displayStatus), VMLocation=location, resourceGroup=strcat('/subscriptions/',subscriptionId,'/resourceGroups/',resourceGroup)
+| order by id asc
+| project id, PowerState, VMLocation, resourceGroup, subscriptionId
+```
+
+<br>
+
+### Query: List of Virtual Machines with their associated disks
+
+This Resource Graph query retrieves a comprehensive list of all Virtual Machines (VMs) in your Azure environment, along with details of their associated disks. It provides insights into the storage configuration of each VM, helping you manage and optimize your storage resources effectively.
+
+<h4>Category</h4>
+
+Resource management
+
+<h4>Query</h4>
+
+```kql
+Resources | where type == "microsoft.compute/virtualmachines"
+| extend osDiskId= tostring(properties.storageProfile.osDisk.managedDisk.id)
+        | join kind=leftouter(resources
+            | where type =~ 'microsoft.compute/disks'
+            | where properties !has 'Unattached'
+            | where properties has 'osType'
+            | project OS = tostring(properties.osType), osSku = tostring(sku.name), osDiskSizeGB = toint(properties.diskSizeGB), osDiskId=tostring(id)) on osDiskId
+        | join kind=leftouter(Resources
+            | where type =~ 'microsoft.compute/disks'
+            | where properties !has "osType"
+            | where properties !has 'Unattached'
+            | project sku = tostring(sku.name), diskSizeGB = toint(properties.diskSizeGB), id = managedBy
+            | summarize sum(diskSizeGB), count(sku) by id, sku) on id
+| project vmId=id, subscriptionId, resourceGroup, OS, location, osDiskId, osSku, osDiskSizeGB, DataDisksGB=sum_diskSizeGB, diskSkuCount=count_sku
+| sort by diskSkuCount desc
```

Comment:
1. Make sure the queries are formatted correctly.
2. You're using a case-insensitive type filter on some lines and case-sensitive on the first where. Which is correct?
3. Let's always use single instead of double quotes.
4. Make sure table names are cased consistently. You have "resources" and "Resources".
5. Doesn't KQL use `order by` instead of `sort by`?
```suggestion
Resources
| where type == 'microsoft.compute/virtualmachines'
| extend osDiskId= tostring(properties.storageProfile.osDisk.managedDisk.id)
| join kind=leftouter(
    Resources
    | where type =~ 'microsoft.compute/disks'
    | where properties !has 'Unattached'
    | where properties has 'osType'
    | project
        OS = tostring(properties.osType),
        osSku = tostring(sku.name),
        osDiskSizeGB = toint(properties.diskSizeGB),
         osDiskId=tostring(id)
) on osDiskId
| join kind=leftouter(
    Resources
    | where type =~ 'microsoft.compute/disks'
    | where properties !has "osType"
    | where properties !has 'Unattached'
    | project
        sku = tostring(sku.name),
        diskSizeGB = toint(properties.diskSizeGB),
        id = managedBy
    | summarize sum(diskSizeGB), count(sku) by id, sku
) on id
| project
    vmId = id,
    subscriptionId,
    resourceGroup,
    OS,
    location,
    osDiskId,
    osSku,
    osDiskSizeGB,
    DataDisksGB = sum_diskSizeGB,
    diskSkuCount = count_sku
| order by diskSkuCount desc
```